### PR TITLE
Raise better error on missing data

### LINKED ISSUE #890: Return empty data instead of error for `get_data()`
Moved from issue in gordo-infrastructure:

@henrikny 
> .get_data() method throws the error One or more series in the tag list is missing data for the specified period
This is kind of annoying when you're trying to fetch data for 50+ tags and you have no idea which one is the culprit
Suggestion:
Option to return empty data

### PR DESCRIPTION
**Pull request by milesgranger**
Will close #890 

Catch individual `IndexError`s during `join_timeseries` and accumulate a list of the series name's that caused the errors and reflect those back out in the final `InsufficientDataError`

### CODE REVIEW

File: gordo/machine/dataset/base.py [hunk 1]
```diff
@@ -115,6 +119,14 @@ def join_timeseries(
 
         """
         resampled_series = []
+        series_iterable = list(series_iterable)
```
Comment (epa095) [inline gordo/machine/dataset/base.py]: Carefull! This loads all the series in memory, and they can be big. The rest of the stuff is intended to be lazy (using generator).

Comment (milesgranger) [reply gordo/machine/dataset/base.py]: ya ya.. trying to think of a nicer way around this. :(

**Comment by ryanjdillon** [comment]
Very nice :)

File: gordo/machine/dataset/base.py [hunk 2]
```diff
@@ -115,77 +119,99 @@ def join_timeseries(
...
         return new_series.dropna()
+
+    @staticmethod
+    def _resample(
+        series: pd.Series,
+        resampling_startpoint: datetime,
+        resampling_endpoint: datetime,
+        resolution: str,
+        aggregation_methods: Union[str, List[str], Callable] = "mean",
+    ):
```
Comment (ryanjdillon) [inline gordo/machine/dataset/base.py:156]: Are there tests that cover this bad boy already?

**Review by ryanjdillon — APPROVED** [review]

File: gordo/machine/dataset/base.py (hunk 2 above)
Comment (milesgranger) [reply gordo/machine/dataset/base.py:156]: A couple in here: https://github.com/equinor/gordo/blob/master/tests/gordo/machine/dataset/test_dataset.py

**Pull request merged**